python/lsst/analysis/tools/analysisMetrics/analysisMetrics.py

```diff
@@ -291,3 +296,49 @@ def setDefaults(self):
             "stdevSky": "nJy",
             "sigmaMADSky": "nJy",
         }
+
+
+class StellarPhotometricRepeatabilityMetric(AnalysisMetric):
+    parameterizedBand: bool = True
+    fluxType: str = "psfFlux"
+
+    def setDefaults(self):
+        super().setDefaults()
+
+        # Apply per-source selection criteria
+        self.prep.selectors.bandSelector = BandSelector()
+
+        # Compute per-group quantities
+        self.process.buildActions.perGroupSn = PerGroupStatistic()
+        self.process.buildActions.perGroupSn.buildAction = Sn(fluxType=f"{self.fluxType}")
+        self.process.buildActions.perGroupSn.func = "median"
+        self.process.buildActions.perGroupExtendedness = PerGroupStatistic()
+        self.process.buildActions.perGroupExtendedness.buildAction.vectorKey = "extendedness"
+        self.process.buildActions.perGroupExtendedness.func = "median"
+        self.process.buildActions.perGroupCount = PerGroupStatistic()
+        self.process.buildActions.perGroupCount.buildAction.vectorKey = f"{self.fluxType}"
+        self.process.buildActions.perGroupStdev = PerGroupStatistic()
+        self.process.buildActions.perGroupStdev.buildAction = MagColumnNanoJansky(vectorKey=f"{self.fluxType}")
+        self.process.buildActions.perGroupStdev.func = "std"
+
+        # Filter on per-group quantities
+        self.process.filterActions.perGroupStdev = DownselectVector(vectorKey="perGroupStdev")
+        self.process.filterActions.perGroupStdev.selector = AndSelector()
+        self.process.filterActions.perGroupStdev.selector.selectors.count = ThresholdSelector(
+            columnKey="perGroupCount", op="ge", threshold=3,
+        )
+        self.process.filterActions.perGroupStdev.selector.selectors.sn = ThresholdSelector(
+            columnKey="perGroupSn", op="ge", threshold=100,
+        )
+        self.process.filterActions.perGroupStdev.selector.selectors.extendedness = ThresholdSelector(
+            columnKey="perGroupExtendedness", op="le", threshold=0.5,
+        )
+
+        self.process.calculateActions.photRepeatStdev = MedianAction(colKey="perGroupStdev")
+
+        self.produce.units = {  # type: ignore
+            "photRepeatStdev": "mag",
+        }
+        self.produce.newNames = {
+            "photRepeatStdev": "{band}_stellarPhotRepeatStdev"
```

Comment:
I see `{band}` here, but I dont see band elsewhere and you are defining psfFlux not per band. where does the per-band calculation come int?

Reply:
We are performing measurements on sources catalogs which have `psfFlux` measurements. Each source is associated with an exposure which is turn is associated with a band, such that each source has a `band` column. In the prep stage, we select only the source measurements in the band of interest.

`self.prep.selectors.bandSelector = BandSelector()`